src/Data/Functor/Extend.hs

```diff
@@ -80,6 +83,24 @@ class Functor w => Extend w where
   {-# MINIMAL duplicated | extended #-}
 #endif
 
+-- | Generic 'duplicated'. Caveats:
+--
+--   1. Will not compile if @w@ is a product type.
+--   2. Will not compile if @w@ is a recursive type.
```

Comment:
Similarly, I can't think of a way to give `(:.:)` an `Extend` instance, so we should just change the wording here a bit.